osc-server/src/main/java/org/osc/core/broker/service/tasks/conformance/openstack/securitygroup/UpdateOrDeleteK8sSecurityGroupMetaTask.java

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ * Copyright (c) Intel Corporation
+ * Copyright (c) 2017
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.osc.core.broker.service.tasks.conformance.openstack.securitygroup;
+
+import javax.persistence.EntityManager;
+
+import org.osc.core.broker.job.TaskGraph;
+import org.osc.core.broker.model.entities.virtualization.SecurityGroup;
+import org.osc.core.broker.service.tasks.TransactionalMetaTask;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+@Component(service = UpdateOrDeleteK8sSecurityGroupMetaTask.class)
+public class UpdateOrDeleteK8sSecurityGroupMetaTask extends TransactionalMetaTask {
+
+    @Reference
+    CheckK8sSecurityGroupLabelMetaTask checkK8sSecurityGroupLabelMetaTask;
+
+    private SecurityGroup sg;
+
+    private TaskGraph tg;
+
+    public UpdateOrDeleteK8sSecurityGroupMetaTask create(SecurityGroup sg) {
+        UpdateOrDeleteK8sSecurityGroupMetaTask task = new UpdateOrDeleteK8sSecurityGroupMetaTask();
+        task.sg = sg;
+        task.checkK8sSecurityGroupLabelMetaTask = this.checkK8sSecurityGroupLabelMetaTask;
+
+        return task;
+    }
+
+    @Override
+    public TaskGraph getTaskGraph() {
+        return this.tg;
+    }
+
+    @Override
+    public String getName() {
+        return "UpdateOrDeleteK8sSecurityGroupMetaTask";
+    }
+
+    @Override
+    public void executeTransaction(EntityManager em) throws Exception {
+        this.tg = new TaskGraph();
+
+        final boolean isDelete = this.sg.getMarkedForDeletion();
+
+        this.sg.getSecurityGroupMembers().forEach(sgm -> {
+            this.tg.addTask(this.checkK8sSecurityGroupLabelMetaTask.create(sgm, isDelete));
+        });
+    }
```

Comment:
After this you must append the existing metatask PortGroupCheckMetaTask. It must run after all these are finished.